src/test/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriterRoundtripTest.java

```diff
@@ -56,124 +61,160 @@ public static Stream<File> data() throws Exception {
+                missingTags.remove((Integer) ExifTagConstants.EXIF_TAG_GPSINFO.tag);
+                missingTags.remove((Integer) ExifTagConstants.EXIF_TAG_INTEROP_OFFSET.tag);
+                missingTags.remove((Integer) TiffTagConstants.TIFF_TAG_JPEG_INTERCHANGE_FORMAT.tag);
+
+                assertTrue(missingTags.isEmpty(), "Missing tags: " + missingTags);
+            }
+
+            for (TiffOutputField sourceField : sourceFields) {
+
+                final boolean isOffsetField =
+                        sourceField.tag == ExifTagConstants.EXIF_TAG_EXIF_OFFSET.tag ||
+                                sourceField.tag == ExifTagConstants.EXIF_TAG_GPSINFO.tag ||
+                                sourceField.tag == ExifTagConstants.EXIF_TAG_INTEROP_OFFSET.tag ||
+                                sourceField.tag == TiffTagConstants.TIFF_TAG_JPEG_INTERCHANGE_FORMAT.tag;
+
+                /*
+                 * Ignore offset fields. They may not be needed after rewrite
+                 * and their value changes anyway.
+                 */
+                if (isOffsetField)
+                    continue;
+
+                TiffOutputField duplicatedField = duplicatedDirectory.findField(sourceField.tag);
+
+                assertNotNull(duplicatedField, "Field is missing: " + sourceField.tagInfo);
+
+                assertEquals(sourceField.tag, duplicatedField.tag, "TiffOutputField tag mismatch.");
+                assertEquals(sourceField.abstractFieldType, duplicatedField.abstractFieldType, "TiffOutputField fieldType mismatch.");
+                assertEquals(sourceField.count, duplicatedField.count, "TiffOutputField count mismatch.");
+
+                boolean sameValue = sourceField.bytesEqual(duplicatedField);
+
+                assertTrue(sameValue, "Bytes are different for field: " + sourceField);
             }
         }
     }
 
     private void copyToDuplicateFile(final File sourceFile, final TiffOutputSet duplicateTiffOutputSet) throws IOException, ImagingException, ImagingException {
+
         final ExifRewriter exifRewriter = new ExifRewriter();
+
         duplicateFile = createTempFile();
+
         try (OutputStream duplicateOutputStream = new BufferedOutputStream(new FileOutputStream(duplicateFile))) {
             exifRewriter.updateExifMetadataLossless(sourceFile, duplicateOutputStream, duplicateTiffOutputSet);
         }
     }
 
     private File createTempFile() {
+
         final String tempDir = FileUtils.getTempDirectoryPath();
         final String tempFileName = this.getClass().getName() + "-" + random.nextLong() + ".tmp";
+
         return new File(tempDir, tempFileName);
     }
 
     private TiffOutputSet duplicateTiffOutputSet(final TiffOutputSet sourceTiffOutputSet) throws ImagingException {
-        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet();
+
+        final TiffOutputSet duplicateTiffOutputSet = new TiffOutputSet(
+                sourceTiffOutputSet.byteOrder
```

Comment:
This was the reason for the orientation value change that @kinow mentioned in https://github.com/apache/commons-imaging/pull/275#pullrequestreview-1305362377

The byte order must be preserved or else the interpretation of bytes will change.